tensorly/decomposition/_parafac2.py

```diff
@@ -48,21 +48,26 @@ def _update_imputed(tensor_slices, mask, decomposition, method):
             slice_mean = tl.sum(slice * slice_mask) / (
                 tl.prod(slice.shape) - len(tl.where(slice_mask == 0)[0])
             )
+
             for i in range(slice.shape[0]):
                 for j in range(slice.shape[1]):
                     if slice_mask[i, j] == 0:
                         tl.index_update(
-                            tensor_slices, tl.index[slice_no, i, j], slice_mean
+                            tensor_slices[slice_no], tl.index[i, j], slice_mean
                         )
```

Comment:
I believe we could use [`tl.where`](https://tensorly.org/stable/modules/generated/tensorly.where.html#tensorly.where) here:

```suggestion
            tensor_slices[slice_no] = tl.where(slice_mask, tensor_slices[slice_no], tensor_slices[slice_no], slice_mean)
```

I'm not entirely sure though (I always need to double check how `where` works), but I think it should work :)

Reply:
Absolutely true. Thanks a lot for bringing this to my attention! I addressed that in a424e3205d5c239e6825c4cdb4a23b63f61f3751.